skrub/_reporting/_table_report.py

```diff
@@ -197,6 +199,36 @@ def _repr_mimebundle_(self, include=None, exclude=None):
     def _repr_html_(self):
         return self._repr_mimebundle_()["text/html"]
 
+    def write_html(self, file):
+        """Store the report into an HTML file.
+
+        Parameters
+        ----------
+        file : str, pathlib.Path or file object
+            The file object or path of the file to store the HTML output.
+        """
+        html = self.html()
+        if isinstance(file, (str, Path)):
+            with open(file, "w", encoding="utf8") as stream:
+                stream.write(html)
+            return
+        try:
+            file.write(html.encode("utf-8"))
+            return
+        except TypeError:
+            pass
+
+        if (encoding := getattr(file, "encoding", None)) is not None:
+            try:
+                assert codecs.lookup(encoding).name == "utf-8"
+            except (AssertionError, LookupError):
+                raise ValueError(
+                    "If `file` is a text file it should use utf-8 encoding; got:"
+                    f" {encoding!r}"
+                )
+
+        file.write(html)
```

Comment:
Could you add a comment explaining what `html` is expected (or not expected) to be at line 230?
Additionally, light inline documentation/comments on the steps above would help readability :)